test/distributed/org/apache/cassandra/distributed/test/sai/ConcurrencyFactorTest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.distributed.test.sai;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import org.apache.cassandra.distributed.Cluster;
+import org.apache.cassandra.distributed.api.ConsistencyLevel;
+import org.apache.cassandra.distributed.impl.TracingUtil;
+import org.apache.cassandra.distributed.test.TestBaseImpl;
+import org.apache.cassandra.utils.TimeUUID;
+
+import static org.apache.cassandra.distributed.api.Feature.GOSSIP;
+import static org.apache.cassandra.distributed.api.Feature.NETWORK;
+import static org.awaitility.Awaitility.await;
+
+public class ConcurrencyFactorTest extends TestBaseImpl
+{
+    private static final String SAI_TABLE = "sai_simple_primary_key";
+
+    private static final int nodes = 3;
+
+    private org.apache.cassandra.distributed.Cluster cluster;
+
+    @Before
+    public void init() throws IOException
+    {
+        cluster = init(Cluster.build(nodes).withTokenSupplier(node -> {
+            switch (node)
+            {
+                case 1: return -9223372036854775808L;
+                case 2: return -3074457345618258602L;
+                case 3: return 3074457345618258603L;
+                default: throw new IllegalArgumentException();
```

Comment:
This version of token supplier seems deprecated. It seems all dtests have moved to use `withTokenSupplier(TokenSupplier.evenlyDistributedTokens(NODES, 1))`. I guess we could use that here too, but that would produce a slightly different distribution, so we might prefer to leave the current deprecated token supplier as it is.

Reply:
I have replaced this with a `TokenSupplier` in the test that supplies the same tokens as above but doesn't use the deprecated interface. I did try the `evenlyDistributedTokens` but it didn't supply the same tokens for the nodes.

Reply:
Nice. The changes have left a few unused imports that break the build.

Reply:
Sure, there are three unused imports in this files that trigger a failure on `checkstyle-test`, which is a dependency of `build-test`:
```
checkstyle-test:
[checkstyle] Running Checkstyle 10.8.1 on 1801 files
[checkstyle] [ERROR] /Users/adelapena/src/cassandra/trunk/test/distributed/org/apache/cassandra/distributed/test/sai/ConcurrencyFactorTest.java:22:8: Unused import - java.util.ArrayList. [UnusedImports]
[checkstyle] [ERROR] /Users/adelapena/src/cassandra/trunk/test/distributed/org/apache/cassandra/distributed/test/sai/ConcurrencyFactorTest.java:23:8: Unused import - java.util.Collection. [UnusedImports]
[checkstyle] [ERROR] /Users/adelapena/src/cassandra/trunk/test/distributed/org/apache/cassandra/distributed/test/sai/ConcurrencyFactorTest.java:30:8: Unused import - com.google.common.collect.Lists. [UnusedImports]
```
CircleCI also hits this failure: https://app.circleci.com/pipelines/github/adelapena/cassandra/3001/workflows/e7d9b183-7cd0-4279-8cf8-0eb53e2c1020/jobs/56300

Maybe you were trying `ant build` or `ant checkstyle`? Those don't include test classes.

Reply:
I see those additional imports are gone in the last changes, they were there yesterday.

Reply:
I must have pushed at some point later because I didn't see them this morning. Anyway, they are gone now.